src/ucp/core/ucp_request.inl

```diff
@@ -176,12 +176,24 @@ UCS_PTR_MAP_IMPL(request, 0);
     }
 
 
+#define UCP_REQUEST_CHECK_OP_ATTR_FLAGS(_param) \
```

Comment:
why need separate macro?

Reply:
The `UCP_REQUEST_CHECK_PARAM` macro is not used by put operation. i'm not sure if it's a mistake or not.
From what I see it uses `ucs_assert` to validate the memory type when calling `ucp_request_get_memory_type`.

You think we should change that?

Reply:
it's a mistake. we need to use UCP_REQUEST_CHECK_PARAM also for put/get/atomic operations